### PR TITLE
[LPT] ConvolutionBackpropData fix FQ weights folding

### PR DESCRIPTION
**Pull request by vzinovie**
### Details:
 - Wrong channels dimension has been chosen for folding in case of not transforming ConvolutionBackpropData.

### Tickets:
 - 59593

### CODE REVIEW

File: inference-engine/src/low_precision_transformations/src/convolution_backprop_data.cpp [hunk 1]
```diff
@@ -63,7 +63,9 @@ bool ConvolutionBackpropDataTransformation::transform(TransformationContext &con
                          NetworkHelper::getDequantization(reshapeFromWeights);
         if (dequantization.empty()) {
             const auto fqOnWeights = getFakeQuantizeOnWeights(convolutionBackpropData);
-            std::shared_ptr<ngraph::Node> resultConstant = NetworkHelper::fold_fake_quantize(fqOnWeights);
+            auto constantShape = fqOnWeights->input(1).get_shape();
+            std::shared_ptr<ngraph::Node> resultConstant =
+                    NetworkHelper::fold_fake_quantize(fqOnWeights, false, constantShape[1] != 1ul ? 1ul : 0ul);
```
Comment (eshoguli) [inline inference-engine/src/low_precision_transformations/src/convolution_backprop_data.cpp:68]:
```suggestion
            std::shared_ptr<ngraph::Node> resultConstant = NetworkHelper::fold_fake_quantize(fqOnWeights, false, 1ul);
```

File: inference-engine/tests/functional/inference_engine/lp_transformations/convolution_backprop_data_transformation.cpp [hunk 2]
```diff
@@ -455,6 +455,26 @@ const std::vector<ConvolutionBackpropDataTransformationTestValues> testValues =
...
+        // ExpectedValues
+        {
+            ngraph::element::u8,
+            {{ngraph::element::f32}, {128.f}, {0.01f}},
+            {},
+            {},
+            op::Constant::create(ngraph::element::f32, ngraph::Shape{}, std::vector<float>{ 0.2f }),
+            true
+        }
+    },
```
Comment (eshoguli) [inline inference-engine/tests/functional/inference_engine/lp_transformations/convolution_backprop_data_transformation.cpp:477]: please update to reproduce the issue

**Review by eshoguli — COMMENTED** [review]
please, update

File: inference-engine/src/low_precision_transformations/src/convolution_backprop_data.cpp (hunk 1 above)
Comment (vzinovie) [reply inference-engine/src/low_precision_transformations/src/convolution_backprop_data.cpp:68]: @eshoguli This check is needed for issue: 56886

File: inference-engine/tests/functional/inference_engine/lp_transformations/convolution_backprop_data_transformation.cpp (hunk 2 above)
Comment (vzinovie) [reply inference-engine/tests/functional/inference_engine/lp_transformations/convolution_backprop_data_transformation.cpp]: Reproduced

**Pull request closed without merging**